Packages/vcs/Lib/vcsvtk/isofillpipeline.py

```diff
@@ -0,0 +1,296 @@
+
+    def _updateContourLevelsAndColors(self):
+        """Overrides baseclass implementation."""
+        # Contour values:
+        self._contourLevels = self._gm.levels
+        if numpy.allclose(self._contourLevels[0], [0., 1.e20]) or \
+                numpy.allclose(self._contourLevels, 1.e20):
+            levs2 = vcs.mkscale(self._scalarRange[0],
+                                self._scalarRange[1])
+            if len(levs2) == 1: # constant value ?
+                levs2 = [levs2[0], levs2[0]+ .00001]
+            self._contourLevels = []
+            if vcs2vtk.isTrue(self._gm.ext_1):
+                ## user wants arrow at the end
+                levs2[0] = -1.e20
+            if vcs2vtk.isTrue(self._gm.ext_2):
+                ## user wants arrow at the end
+                levs2[-1] = 1.e20
+            for i in range(len(levs2) - 1):
+                self._contourLevels.append([levs2[i], levs2[i+1]])
+        else:
+            if not isinstance(self._gm.levels[0], (list, tuple)):
+                self._contourLevels = []
+                levs2 = self._gm.levels
+                if numpy.allclose(levs2[0], 1.e20):
+                    levs2[0] = -1.e20
+                for i in range(len(levs2) - 1):
+                    self._contourLevels.append([levs2[i], levs2[i+1]])
+
+        if isinstance(self._contourLevels, numpy.ndarray):
+            self._contourLevels = self._contourLevels.tolist()
+
+        ## Figure out colors
+        self._contourColors = self._gm.fillareacolors
+        if self._contourColors == [1,]:
+            # TODO BUG It's possible that levs2 may not exist here...
+            self._contourColors = vcs.getcolors(levs2, split=0)
+            if isinstance(self._contourColors, (int, float)):
+                self._contourColors = [self._contourColors,]
+
+    def _createPolyDataFilter(self):
+        """Overrides baseclass implementation."""
+        self._vtkPolyDataFilter = vtk.vtkDataSetSurfaceFilter()
+        if self._useCellScalars:
+            # Sets data to point instead of just cells
+            c2p = vtk.vtkCellDataToPointData()
+            c2p.SetInputData(self._vtkDataSet)
+            c2p.Update()
+            #For contouring duplicate points seem to confuse it
+            # TODO Charles:
+            # There has been a typo in the "IsA" for a long time. When I fix
+            # it, tests start failing as vtkCleanUnstructuredGrid is a
+            # ParaView filter that is not currently available in VTK.
+            # There haven't been any problems from skipping this step in the
+            # tests, is it really needed?
+            if self._vtkDataSet.IsA("vtkUntructuredGrid"):
+                cln = vtk.vtkCleanUnstructuredGrid()
+                cln.SetInputConnection(c2p.GetOutputPort())
+                self._vtkPolyDataFilter.SetInputConnection(cln.GetOutputPort())
+            else:
+                self._vtkPolyDataFilter.SetInputConnection(c2p.GetOutputPort())
```

Comment:
@doutriaux1 See the comment/typo above. The true branch of this conditional has never executed since there is no class in VTK matching the `IsA` argument. This hasn't cause problems so far, any objections to me removing this check and just leaving the `else` body?